_studio/shared/src/mfx_mfe_adapter.cpp

```diff
@@ -329,28 +305,12 @@ mfxStatus MFEVAAPIEncoder::Submit(VAContextID context, vm_tick timeToWait, bool
 
     if(!framesToSubmit)
     {
-        vm_mutex_unlock(&m_mfe_guard);
         return MFX_ERR_UNDEFINED_BEHAVIOR;
     }
-    vm_tick start_tick = vm_time_get_tick();
-    vm_tick spent_ticks = 0;
-    timeout = timeout? timeout: cur_stream->timeout;
-    while (m_framesCollected < framesToSubmit &&
-           !cur_stream->isFrameSubmitted() &&
-           timeToWait > spent_ticks)
-    {
-        vm_status res = vm_cond_timed_uwait(&m_mfe_wait, &m_mfe_guard,
-                                          (timeToWait - spent_ticks));
-        if ((VM_OK == res) || (VM_TIMEOUT == res))
-        {
-            spent_ticks = (vm_time_get_tick() - start_tick);
-        }
-        else
-        {
-            vm_mutex_unlock(&m_mfe_guard);
-            return MFX_ERR_UNKNOWN;
-        }
-    }
+    m_mfe_wait.wait_for(guard, std::chrono::microseconds(timeToWait), [this, &framesToSubmit, &cur_stream] {
```

Comment:
Wait... In which values we have timeToWait? I thought it is in milliseconds. In this case std::chrono::milliseconds(timeToWait) should be used otherwise you will treat timeToWait as microseconds and eventually wait much shorter than you should. @artem-shaporenko : can you, please, comment?

Reply:
if you look at initial interface there is a comment:
//time passed in vm_tick, so milliseconds to be multiplied by vm_frequency/1000

Reply:
yep, I missed that. Thank you. @akokoshn : does that mean that validation we started is wrong for MFE?

Reply:
@dvrogozh, to be honest I'm not sure, but  probably yes. Please correct me if I wrong, it's possible to rerun MFE cases, if validation already started. Any way I will update test_system.

Reply:
@akokoshn why you captured framesToSubmit and cur_stream by reference? They are trivially copyable, so capturing by value fits better